bitnami/grafana/templates/deployment.yaml

```diff
@@ -0,0 +1,172 @@
+    spec:
+      {{- if .Values.schedulerName }}
+      schedulerName: "{{ .Values.schedulerName }}"
+      {{- end }}
+{{- include "grafana.imagePullSecrets" . | indent 6 }}
+     {{- if .Values.securityContext.enabled }}
+      securityContext:
+        runAsUser: {{ .Values.securityContext.runAsUser }}
+        fsGroup: {{ .Values.securityContext.fsGroup }}
+        runAsNonRoot: {{ .Values.securityContext.runAsNonRoot }}
+      {{- end }}
+      containers:
+      - name: {{ .Chart.Name }}
+        image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+        imagePullPolicy: {{ .Values.image.pullPolicy }}
+        envFrom:
+        - configMapRef:
+            name: "{{ include "grafana.fullname" . }}-envvars"
+        - secretRef:
+            name: "{{ include "grafana.fullname" . }}-secret"
+        {{- if .Values.smtp.existingSecret }}
+        - name: GF_SMTP_USER
+          valueFrom:
+            secretKeyRef:
+              name: {{ .Values.smtp.existingSecret }}
+              key: {{ .Values.smtp.existingSecretUserKey | default "user" }}
+        - name: GF_SMTP_PASSWORD
+          valueFrom:
+            secretKeyRef:
+              name: {{ .Values.smtp.existingSecret }}
+              key: {{ .Values.smtp.existingSecretPasswordKey | default "password" }}
+        {{- end }}
+{{- if .Values.extraEnvVars }}
+{{ toYaml .Values.extraEnvVars | indent 8 }}
+{{- end }}
+        volumeMounts:
+        {{- if .Values.config.useGrafanaIniFile }}
+        - name: grafana-ini
+          mountPath: /opt/bitnami/grafana/conf/grafana.ini
+          subPath: grafana.ini
+        {{- end }}
+        {{- if .Values.config.useCustomIniFile }}
+        - name: custom-ini
+          mountPath: /opt/bitnami/grafana/conf/custom.ini
+          subPath: custom.ini
+        {{- end }}
+        - name: data
+          mountPath: /opt/bitnami/grafana/data
+        {{- if .Values.dashboardsProvider.enabled }}
+        - name: dashboards-provider
+          mountPath: /opt/bitnami/grafana/conf/provisioning/dashboards
+        {{- end }}
+        {{- range .Values.dashboardsConfigMaps }}
+        - name: {{ . }}
+          mountPath: /opt/bitnami/grafana/dashboards
+        {{- end }}
+        {{- if .Values.datasources.secretName }}
+        - name: datasources
+          mountPath: /opt/bitnami/grafana/conf/provisioning/datasources
+        {{- end }}
+        {{- if and .Values.ldap.enabled .Values.ldap.configMapName }}
+        - name: ldap
+          mountPath: /opt/bitnami/grafana/conf/ldap.toml
+          subPath: ldap.toml
+        {{- end }}
+        {{- range .Values.extraConfigmaps }}
+        - name: {{ .name }}
+          mountPath: {{ .mountPath }}
+          subPath: {{ .subPath | default "" }}
+          readOnly: {{ .readOnly }}
+        {{- end }}
+        ports:
+          - name: dashboard
+            containerPort: 3000
+            protocol: TCP
+        livenessProbe:
+          tcpSocket:
+            port: dashboard
```

Comment:
`tcpSocket` is a very basic probe. Can we use `httpGet` instead? We should at least check the UI returns a `200 OK` code